src/api/control/member.rs

```diff
@@ -0,0 +1,90 @@
+//! Member definitions and implementations.
+
+use hashbrown::HashMap;
+
+use crate::log::prelude::*;
+
+/// ID of [`Member`].
+pub type Id = u64;
+
+/// Media server user with its ID and credentials.
+#[derive(Clone, Debug)]
+pub struct Member {
+    /// ID of [`Member`].
+    pub id: Id,
+    /// Credentials to authorize [`Member`] with.
+    pub credentials: String,
+}
+
+/// Repository that stores [`Member`]s.
+#[derive(Default)]
+pub struct MemberRepository {
+    members: HashMap<Id, Member>,
+}
+
+impl MemberRepository {
+    /// Creates new [`Member`]s repository with passed-in [`Member`]s.
+    pub fn new(members: HashMap<Id, Member>) -> Self {
+        MemberRepository { members }
+    }
+
+    /// Returns [`Member`] by its ID.
+    pub fn get(&self, id: Id) -> Option<Member> {
+        debug!("retrieve member by id: {}", id);
+        self.members.get(&id).map(|member| member.clone())
+    }
+
+    /// Returns [`Member`] by its credentials.
+    pub fn get_by_credentials(&self, credentials: String) -> Option<Member> {
```

Comment:
@Kirguir you're not going to modify `credentials` anyhow inside method, so you don't need to own it. Just borrow via `&str`.